packages/tldraw/src/lib/shapes/geo/GeoShapeUtil.tsx

```diff
@@ -848,40 +853,52 @@ export class GeoShapeUtil extends BaseBoxShapeUtil<TLGeoShape> {
 	}
 }
 
+const scaledLabelSizes: Record<
+	string,
+	Record<
+		string,
+		BoxModel & {
+			scrollWidth: number
+		}
+	>
+> = {}
+
 function getUnscaledLabelSize(editor: Editor, shape: TLGeoShape) {
 	const { text, font, size, w } = shape.props
 
 	if (!text) {
 		return { w: 0, h: 0 }
 	}
 
-	const minSize = editor.textMeasure.measureText('w', {
-		...TEXT_PROPS,
-		fontFamily: FONT_FAMILIES[font],
-		fontSize: LABEL_FONT_SIZES[size],
-		maxWidth: 100, // ?
-	})
-
-	// TODO: Can I get these from somewhere?
-	const sizes = {
-		s: 2,
-		m: 3.5,
-		l: 5,
-		xl: 10,
+	// Cache the minimum size for each font/size combo
+	let minSize = scaledLabelSizes[font]?.[size]
+	if (!minSize) {
+		if (!scaledLabelSizes[font]) {
+			scaledLabelSizes[font] = {}
+		}
+		minSize = editor.textMeasure.measureText('W', {
+			...TEXT_PROPS,
+			fontFamily: FONT_FAMILIES[font],
+			fontSize: LABEL_FONT_SIZES[size],
+			maxWidth: 100, // ?
+		})
+		scaledLabelSizes[font][size] = minSize
```

Comment:
We were measuring the W character every time. Measurement is one of the slower things, so let's just cache a little bit instead.